File: tiq-e/others/1/main.ts
import {performance} from "perf_hooks";

function hammingWeight(n: number): number {
    let sum = 0;
    while (n != 0) {
        sum += n & 1;
        n = n >>> 1;
    }
    return sum;
}

const n = 3;

const start = performance.now();
const result = hammingWeight(n);

const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);

File: tiq-e/others/1/main.go
package main

import (
	"fmt"
	"time"
)

func hammingWeight(num uint32) int {
	res := 0
	for num > 0 {
		if num&1 == 1 {
			res++
		}
		num = num >> 1
	}
	return res
}

func main() {
	n := uint32(3)

	start := time.Now()
	result := hammingWeight(n)

	end := time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
